customer/src/containers/queue.js

```diff
@@ -0,0 +1,86 @@
+import React, { useEffect, forwardRef } from 'react';
+import io from 'socket.io-client';
+
+import Footer from '../ui/footer';
+import MenuItem from '../ui/menuItem';
+import BeerAnimation from '../ui/beerAnimation';
+import PrimaryHead from '../ui/primaryHead';
+import SecondaryHead from '../ui/secondaryHead';
+
+import '../styles/containers/queue.css';
+
+const Queue = forwardRef((props, ref) => {
+  const {
+    order,
+    orderId,
+    orderStatus,
+    updatePage,
+    clearOrder,
+    isMenuOpen,
+    orderSpecialWishes,
+    updateStatus,
+    barId,
+  } = props;
+
+  let socket;
+
+  const closeSocket = () => {
+    socket.removeAllListeners();
+    socket.close();
+  };
+
+  useEffect(() => {
+    socket = io(`/${props.barId}`, {
```

Comment:
Create a Socket module that handles all communications with this channel, this way you abstract the Component from the communication protocol (separation of concerns).
```js
socket = Socket.newConnection(barId, orderId);
```